graph-builder/src/config/settings.rs

```diff
@@ -0,0 +1,92 @@
+use super::Merge;
+use super::{cli, file};
+use failure::Fallible;
+use std::collections::HashSet;
+use std::net::{IpAddr, Ipv4Addr};
+use std::path::PathBuf;
+use std::time;
+use structopt::StructOpt;
+
+/// Runtime application settings (validated config).
+#[derive(Debug, SmartDefault)]
+pub struct AppSettings {
+    /// Listening address for the main service.
+    #[default(IpAddr::V4(Ipv4Addr::LOCALHOST))]
+    pub address: IpAddr,
+
+    /// Optional auth secrets for the registry scraper.
+    #[default(None)]
+    pub credentials_path: Option<PathBuf>,
+
+    /// Required client parameters for the main service.
+    #[default(_code = "HashSet::new()")]
+    pub mandatory_client_parameters: HashSet<String>,
+
+    /// Metadata key where to record the manifest-reference.
+    #[default("com.openshift.upgrades.graph.release.manifestref")]
+    pub manifestref_key: String,
+
+    /// Endpoints namespace for the main service.
+    #[default("")]
```

Comment:
`Default::default()`?

Reply:
Here and above, `Default::default()` is implicit if I remove the annotation (but I liked it explicit). Do you prefer me to drop all the non-custom default annotations?

Reply:
Discussed out-of-band: let's use the implicit form.